### DISCUSSION TITLE
Allow passing variables to new scene using argument

### DISCUSSION THREAD

**Discussion by shak2** [godotengine/godot-proposals#9875] [Engine Core]
Example: change_scene_to_file(SCENEFILE, arg1, arg2, ...)

Why not let users change scene to file with additional arg1 and arg 2 etc.?

Right now the recommendation is to use autoloads, which feels very hacky and messy. 

Remember encapsulation.

Thoughts?

**Comment by AThousandShips** [comment]
Where would the arguments go? What method would be called and where?

**Reply by sairam4123** [reply]
Maybe we could simply build an association between scene and scripts. And when a new scene gets instantiated, a new function called _init_scene will be called with the argument passed when instantiating a scene.

Example:
```
PackedScene.instanciate(arg1, arg2)

# packed_scene.gd

func _init_scene(arg1, arg2):
    pass
```

**Reply by AThousandShips** [reply]
So what happens if the arguments don't match or there isn't such a method?

**Comment by dalexeev** [comment]
* Related: #1513.
